updated readme to include new modes (#53)
maticnetwork/avail-light: README.md

## Introduction

Naive approach for building one AVAIL light client, which will do following

- Listen for newly mined blocks

> - As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.

Author: yeh I will update that also

### Modes of Operation

> 1. **Light client**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.

Reviewer: ```suggestion
1. **Light Client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```

> 2. **App-Specific-Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 

Reviewer: ```suggestion
2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 
```

> 3. **Fat-Client-Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.

Reviewer: ```suggestion
3. **Fat-Client Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.
```

## Introduction

> Naive approach for building one AVAIL light client, which will do following

Reviewer 2: @kroos47 I think the doc team changed the wordings here. Kindly keep the changes proposed by them.

- Listen for newly mined blocks

> - As soon as new block is available, attempts to eventually gain confidence by asking for proof from full client _( via JSON RPC interface )_ for `N` many cells where cell is defined as `{row, col}` pair. The number of cells are randomly sampled to fetch the confidence eventually.

Reviewer 2: Finalized block, right?

### Modes of Operation

> 1. **Light-client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.

Reviewer 2: `gets random cells with proofs` -> It gets the cells it asked for.

> 2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 

Reviewer 2: Maybe we can simplify and say that the relevant cells are downloaded and the data is reconstructed.

`avail-light` is a data availability light client that can do the following:

* Listen for newly forged blocks
* Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.

### Modes of Operation

> 1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculate the confidence.

Reviewer: ```suggestion
1. **Light Client Mode**: The basic mode of operation and is always active in whichever mode is operational. If an `App_ID` is not provided, this mode will commence. The client on each header it receives will do random sampling using RPC calls. It gets random cells with proofs in return, which verifies and calculates the confidence.
```

> 2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode and reconstruct the app_extrinsics_data. 

Reviewer: ```suggestion
2. **App-Specific Mode**: If an **`App_ID` > 0** is given in the config file, the client finds out the `cols` related to the provided `App_ID` using `app_data_lookup` in the header. The client then fetches 50% of cells from `cols`, verifies them, and uses them to decode and reconstruct the `app_extrinsics_data`. 
```

> 3. **Fat-Client-Mode**: In this mode, the client gets the entire extended matrix using IPFS if it is available or uses RPC calls to fetch. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstruct the app_specific_data related to all App_IDs.

Reviewer: ```suggestion
3. **Fat-Client Mode**: The client retrieves the entire extended matrix using IPFS (if available) or fetches via RPC calls. It verifies all the cells and computes the CID mapping for the IPFS Pinning. It then decodes the extended matrix and reconstructs the `app_specific_data` related to all `App_IDs`.
```

`avail-light` is a data availability light client that can do the following:

> * Listen for newly mined blocks

Reviewer 2: I would prefer `forged` over `mined`. However, I think right now we listen to `finalized` blocks, right?

* Gain confidence for `N` *cells*, where *cell* is defined as a `{row, col}` pair.
  > As soon as a new block is available, the light client attempts to gain confidence by asking for a proof from a full client via JSON-RPC.

> * Batch processces blocks in reverse order (i.e. prioritizes the latest blocks) for lower numbered blocks 	where no confidence is yet gained.

Reviewer 2: This is currently happening on v1?

Author: oh rebasing did a mistake 😓  will do the correction

### Modes of Operation

> 1. **Light client**: This is the basic mode of operation and is always active in whichever mode it works. If the APP_ID is not given then this mode will be commence. In this mode, the client on each header it receieves will do the random sampling using RPC calls. It gets random cells in return, which verifies and calculate the confidence.

Reviewer 2: `will commence`
`gets proofs in return`
Some more proofreading required. Docs team can help here.

> 2. **App-Specific-Mode**: In this mode if an App_ID(greater than 0) is given in the config file then the client would find out the cols related the App_ID given using the app_data_lookup in the header. Then it fetches the 50% of cells from the cols, verifies them and uses them to decode the app_extrinsics_data. 

Reviewer 2: Does the decoding work in v1?

Author: yes!! reconstruted data is given in both app and fat mode

> 3. **Fat-Client-Mode**: In this mode, the client gets the entire extended matrix using IPFS if it is available or uses RPC calls to fetch. It verifies the entire cells and computes the CID mapping for the IPFS Pinning. It then decodes the entire matrix and reconstruct the entire app_specific_data related to all App_IDs.

Reviewer 2: `entire cells` -> `all the cells` or `entire matrix`? 
Decodes matrix -> reconstructs original matrix or decodes extended matrix?

Author: I think its original matrix

Author: @aterentic-ethernal

ipfs_port = 37000
ipfs_path = "avail_ipfs_store"

# put full_node_rpc = https://devnet-avail.polygon.technology/ incase you are connecting to devnet

> full_node_rpc = "http://127.0.0.1:9933"

Reviewer 2: The config should take an array, right? Same for `ws`? @kroos47
